packages/mongodb-log-writer/src/mongo-log-manager.ts

```diff
@@ -73,8 +79,11 @@ export class MongoLogManager {
       if (Date.now() - deletionStartTimestamp > maxDurationMs) break;
 
       if (!dirent.isFile()) continue;
-      const { id } =
-        /^(?<id>[a-f0-9]{24})_log(\.gz)?$/i.exec(dirent.name)?.groups ?? {};
+      const logRegExp = new RegExp(
+        `^${this.prefix}(?<id>[a-f0-9]{24})_log(\\.gz)?$`,
```

Comment:
If we do insert the prefix into the regexp, we should either escape it or validate that it does not contain any characters that would have special meaning when interpreted as a regexp in the `MongoLogManager` constructor

Reply:
added regex validation at point of setting the prefix that just limits it to alphanumeric and _